sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -92,7 +92,9 @@ internal class NextLinkOperationImplementation : IOperation
 
         public async ValueTask<OperationState> UpdateStateAsync(bool async, CancellationToken cancellationToken)
```

Comment:
The name of this isn't ideal, but it's been around so there's no value in changing it now.

Reply:
I agree that it would be preferable to follow patterns in Azure.Core to provide one UpdateState and one UpdateStateAsync method, but I don't think it's worth blocking the PR on this.